docs/_posts/examples/3400-01-31-animate-a-line.html

```diff
@@ -0,0 +1,128 @@
+--- layout: example category: example title: Animate a line description: Animate a line by updating a GeoJSON source on each frame. tags: - layers - sources ---
+<style>
+button {
+  position: absolute;
+  margin: 20px;
+}
+
+#play::after {
+  content: 'Stop';
+}
+
+#play.pause::after {
+  content: 'Play';
+}
+</style>
+
+<div id='map'></div>
+<button id='play' class='pause'></button>
+
+<script>
+mapboxgl.accessToken = 'pk.eyJ1IjoieXVuamllbGkiLCJhIjoiY2lxdmV5MG5rMDAxNmZta3FlNGhyMmpicSJ9.CTEQgAyZGROcpJouZuzJyA';
+var map = new mapboxgl.Map({
+  container: 'map',
+  style: 'mapbox://styles/mapbox/streets-v8',
+  center: [0, 0],
+  zoom: 1
+});
+
+// Create a GeoJSON source with an empty lineString.
+var geojson = {
+  "type": "FeatureCollection",
+  "features": [{
+    "type": "Feature",
+    "geometry": {
+      "type": "LineString",
+      "coordinates": [
+        [0, 0]
+      ]
+    }
+  }]
+};
+
+map.on('load', function() {
+
+  // Add a source and layer displaying the line which will be
+  // animated in a circle as a sine wave along the map.
+  map.addSource('line-animation', {
+    'type': 'geojson',
+    'data': geojson
+  });
+
+  map.addLayer({
+    'id': 'line-animation',
+    'type': 'line',
+    'source': 'line-animation',
+    'layout': {
+      'line-cap': 'round',
+      'line-join': 'round'
+    },
+    'paint': {
+      'line-color': '#3cc',
+      'line-width': 5,
+      'line-opacity': .4
+    }
```

Comment:
This is hard to see on the map -- I would make it a brighter color or more opaque and/or change the basemap.